Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionSidebar.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(WK_WEB_EXTENSIONS_SIDEBAR)
+
+#include "APIObject.h"
+#include "CocoaImage.h"
+#include <wtf/Forward.h>
+#include <wtf/WeakPtr.h>
+#include <wtf/text/WTFString.h>
+
+OBJC_CLASS WKWebView;
+
+namespace WebKit {
+
+class WebExtensionContext;
+class WebExtensionTab;
+class WebExtensionWindow;
+
+class WebExtensionSidebar : public API::ObjectImpl<API::Object::Type::WebExtensionSidebar>, public CanMakeWeakPtr<WebExtensionSidebar> {
+    WTF_MAKE_NONCOPYABLE(WebExtensionSidebar);
+
+public:
+    enum class IsDefaultSidebar { No, Yes };
+
+    template<typename... Args>
+    static Ref<WebExtensionSidebar> create(Args&&... args)
+    {
+        return adoptRef(*new WebExtensionSidebar(std::forward<Args>(args)...));
+    }
+
+    explicit WebExtensionSidebar(WebExtensionContext&, IsDefaultSidebar = IsDefaultSidebar::No);
+    explicit WebExtensionSidebar(WebExtensionContext&, WebExtensionTab&);
+    explicit WebExtensionSidebar(WebExtensionContext&, WebExtensionWindow&);
+
+    bool operator==(const WebExtensionSidebar&) const;
+
+    std::optional<Ref<WebExtensionContext const>> extensionContext() const;
```

Comment:
I tried to write both const and non-const getters for context, window, and tab. The non-const getter drops the const within ref. I think that since this method is a const method, and since context is a property of this class, we would need this getter to return with the interior const right?